### PR TITLE
[MINOR][SS][DOCS] Update Structured Streaming guide doc and update code typo

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This is a minor change to update structured-streaming-programming-guide and typos in code. 

### Why are the changes needed?

Keep the user-facing document correct and updated.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 1]
```diff
@@ -225,8 +225,8 @@ object UnsupportedOperationChecker extends Logging {
                 "mapGroupsWithState is not supported with " +
                   s"$outputMode output mode on a streaming DataFrame/Dataset")
             }
-          } else {                                           // check latMapGroupsWithState
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:228]: This is the typo?

File: docs/structured-streaming-programming-guide.md [hunk 2]
```diff
@@ -1777,7 +1779,7 @@ Here is the compatibility matrix.
     <td style="vertical-align: middle;">Update operation mode</td>
     <td style="vertical-align: middle;">Update</td>
     <td style="vertical-align: middle;">
-      Aggregations not allowed after <code>flatMapGroupsWithState</code>.
+      Aggregations not allowed in a query with <code>flatMapGroupsWithState</code>.
```
Comment (viirya) [inline docs/structured-streaming-programming-guide.md:1782]: Aggregations are not just disallowed after `flatMapGroupsWithState` in in update mode. It is disallowed in the query.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 3]
```diff
@@ -252,7 +252,7 @@ object UnsupportedOperationChecker extends Logging {
               } else if (collectStreamingAggregates(m).nonEmpty) {
                 throwError(
                   "flatMapGroupsWithState in append mode is not supported after " +
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:254]: Question. This `after` looks correct, doesn't it?

File: docs/structured-streaming-programming-guide.md [hunk 4]
```diff
@@ -1763,7 +1763,9 @@ Here is the compatibility matrix.
   <tr>
     <td colspan="2" style="vertical-align: middle;">Queries with <code>mapGroupsWithState</code></td>
     <td style="vertical-align: middle;">Update</td>
-    <td style="vertical-align: middle;"></td>
+    <td style="vertical-align: middle;">
+      Aggregations not allowed in a query with <code>mapGroupsWithState</code>.
+    </td>
```
Comment (viirya) [inline docs/structured-streaming-programming-guide.md:1768]: `UnsupportedOperationChecker.scala`:

```scala
if (m.isMapGroupsWithState) {                       // check mapGroupsWithState
  // allowed only in update query output mode and without aggregation
  if (aggsInQuery.nonEmpty) {
    throwError(
      "mapGroupsWithState is not supported with aggregation " +
      "on a streaming DataFrame/Dataset")
  } else if (outputMode != InternalOutputModes.Update) {
    ...
  }
} ..
..
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala (hunk 1 above)
Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:228]: yes

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala (hunk 3 above)
Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:254]: yes.

Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:254]: This is append mode.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @viirya .

**Comment by viirya** [comment]
Thanks @dongjoon-hyun

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 5]
```diff
@@ -212,11 +212,11 @@ object UnsupportedOperationChecker extends Logging {
         case m: FlatMapGroupsWithState if m.isStreaming =>
 
           // Check compatibility with output modes and aggregations in query
-          val aggsAfterFlatMapGroups = collectStreamingAggregates(plan)
+          val aggsInQuery = collectStreamingAggregates(plan)
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala:215]: Here it doesn't only collect aggregations after `FlatMapGroupsWithState`, but all aggregations in  the query.

**Comment by HyukjinKwon** [comment]
cc @HeartSaVioR and @xuanyuanking FYI

**Review by HyukjinKwon — APPROVED** [review]

**Comment by xuanyuanking** [comment]
LGTM, thanks for fixing.

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**